### PR TITLE
Add axis=1 for concat

### PR DESCRIPTION
**Pull request by charlesdong1991**
#625

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Nice! Thanks for working on this!

**Comment by charlesdong1991** [comment]
haven't come up with a nice and robust solution for allowing `ingore_index` to be `True`, any suggestions? @ueshin

File: databricks/koalas/namespace.py [hunk 1]
```diff
@@ -1488,17 +1497,40 @@ def concat(objs, axis=0, join='outer', ignore_index=False):
...
+                obj = obj.rename('0').to_dataframe()
+            else:
+                if obj.name == '0':
+                    obj = obj.rename(str(idx)).to_dataframe()
+                else:
+                    obj = obj.to_dataframe()
         new_objs.append(obj)
     objs = new_objs
 
+    if axis == 1:
```
Comment (HyukjinKwon) [inline databricks/koalas/namespace.py]: For `ignore_index`, I think we can explicitly throw an exception for now when axis is 1.

File: databricks/koalas/namespace.py [hunk 2]
```diff
@@ -1472,8 +1477,12 @@ def concat(objs, axis=0, join='outer', ignore_index=False):
                         'objects, you passed an object of type '
                         '"{name}"'.format(name=type(objs).__name__))
 
-    if axis not in [0, 'index']:
-        raise ValueError('axis should be either 0 or "index" currently.')
+    if axis in [0, 'index']:
+        axis = 0
+    elif axis in [1, 'columns']:
+        axis = 1
```
Comment (HyukjinKwon) [inline databricks/koalas/namespace.py]: Seems conflict was by https://github.com/databricks/koalas/commit/6ece69d031f7d3932e93c3a5883a3b4e78d36b59

**Comment by charlesdong1991** [comment]
Many thanks for your comments @HyukjinKwon 

I am very sorry that I have been very busy recently and don't have time for my existing PRs. I will try my best to update the PR following your comments this weekend!

Thanks again for your understanding and patience!

**Comment by HyukjinKwon** [comment]
Hey, @charlesdong1991 no worries at all. Thanks for trying this change out.

File: databricks/koalas/namespace.py [hunk 3]
```diff
@@ -1488,17 +1497,40 @@ def concat(objs, axis=0, join='outer', ignore_index=False):
...
+        cols = []
+        for idx, obj in enumerate(objs):
+            cols.extend(obj._internal.data_columns)
+            if idx == 0:
+                internal = obj._internal
+                index_col = internal.index_columns[0]
+                sdfs = internal._sdf
+                continue
+            col = '{}'.format(index_col)
+            sdfs = sdfs.join(obj._internal._sdf,
```
[the remark this answers is not in the document]
Comment (charlesdong1991) [reply databricks/koalas/namespace.py]: Very sorry to wait such long to get back to this PR @HyukjinKwon

I am not sure if i fully understand here, if via each column assignment, doesn't it mean the ops become more expensive? I made a slight change, feel free to review and will appreciate your feedback and suggestions! I got through tough time already, so will have time to make code changes much quicker.

**Pull request closed without merging**